### PR TITLE
Use InsecureSkipVerify only with https upstreams

### PR DESCRIPTION
**Pull request by peusebiu**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/sync/utils.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func getHTTPClient(regCfg *RegistryConfig, credentials Credentials, log log.Logg
 	}
 
 	// nolint: gosec
-	if regCfg.TLSVerify != nil && !*regCfg.TLSVerify {
+	if regCfg.TLSVerify != nil && !*regCfg.TLSVerify && strings.HasPrefix(regCfg.URL, "https") {
```
Comment (rchincha) [inline pkg/extensions/sync/utils.go:136]: What if "httpsabc.weirdprefix.com"?
Would actually parse the uri and then .Scheme

Comment (peusebiu) [reply pkg/extensions/sync/utils.go:136]: Ok, I'll use an url parser

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Comment by rchincha** [comment]
Rebase and resubmit

**Review by rchincha — APPROVED** [review]

**Pull request merged**